nmagent/responses.go

```diff
@@ -37,8 +43,74 @@ type NCVersionList struct {
 	Containers []NCVersion `json:"networkContainers"`
 }
 
+// HomeAZFix is an indication that a particular bugfix has been applied to some
+// HomeAZ.
+type HomeAZFix int
+
+func (h HomeAZFix) String() string {
+	switch h {
+	case HomeAZFixInvalid:
+		return "HomeAZFixInvalid"
+	case HomeAZFixIPv6:
+		return "HomeAZFixIPv6"
+	default:
+		return "Unknown HomeAZ Fix"
+	}
+}
+
+const (
+	HomeAZFixInvalid HomeAZFix = iota
+	HomeAZFixIPv6
+)
+
 type AzResponse struct {
-	HomeAz uint `json:"homeAz"`
+	HomeAz       uint
+	AppliedFixes []HomeAZFix
+}
+
+func (az *AzResponse) UnmarshalJSON(in []byte) error {
+	type resp struct {
+		HomeAz     uint `json:"homeAz"`
+		APIVersion uint `json:"apiVersion"`
+	}
+
+	var rsp resp
+	err := json.Unmarshal(in, &rsp)
+	if err != nil {
+		return errors.Wrap(err, "unmarshaling raw home az response")
+	}
+
+	if rsp.APIVersion != 0 && rsp.APIVersion != 2 {
+		return HomeAzAPIVersionError{
+			ReceivedAPIVersion: rsp.APIVersion,
+		}
+	}
+
+	az.HomeAz = rsp.HomeAz
+
+	if rsp.APIVersion == 2 { // nolint:gomnd // ignore magic number 2
+		az.AppliedFixes = append(az.AppliedFixes, HomeAZFixIPv6)
+	}
+
+	return nil
+}
+
+// ContainsFixes reports whether all fixes requested are present in the
+// AzResponse returned.
+func (az AzResponse) ContainsFixes(requestedFixes ...HomeAZFix) bool {
+	for _, requested := range requestedFixes {
+		found := false
+		for _, present := range az.AppliedFixes {
+			if requested == present {
+				found = true
+			}
+		}
+
+		if !found {
+			return false
+		}
+	}
+	return true
```

Comment:
Can get rid of all this code if we just change it to uint and take decisions based on 0 or 2 value. Anything other than that is invalidated already.

Reply:
I think I've answered this in the other comment, but here's another example:

```go
// do something if the apiVersion indicates that the ipv6 fix was applied
//nolint:gomnd // see nmagent documentation for these values
switch response.apiVersion {
       case 2:
              // do something here
       case 0:
              // do something else
       default:
              return fmt.Errorf("unexpected apiVersion value %d", response.apiVersion)
}
```
vs.

```go
if response.ContainsFixes(nmagent.HomeAZFixIPv6) {
       // do something here
}
```

The latter is self-documenting code, doesn't require dealing with invalid values (easy to forget), and doesn't require disabling a common linter all the time. It's also more flexible, as it supports detecting N fixes.

Reply:
Sounds reasonable.